src/kernel/src/cmdline/mod.rs

```diff
@@ -131,24 +131,31 @@ impl Cmdline {
         valid_element(v)?;
         self.has_capacity(k.len() + v.len() + 1)?;
 
-        self.start_push();
-        self.line.push_str(k);
-        self.line.push('=');
-        self.line.push_str(v);
+        if let Some(index) = self.line.find("--") {
+            self.line.insert_str(index, &format!("{}={} ", k, v))
+        } else {
+            self.start_push();
+            self.line.push_str(&format!("{}={}", k, v))
+        }
         self.end_push();
 
         Ok(())
     }
 
-    /// Validates and inserts a string to the end of the current command line.
+    /// Validates and inserts a string into the commandline. The position of the
+    /// string depends on the presence of  "--".
```

Comment:
I would also specify here that string is inserted before `--`, if present. It's not clear from the description.